lib/Log/Any/Adapter/DERIV.pm

```diff
@@ -354,87 +386,117 @@ The log data.
 
 =back
 
-Return: processed data
-
+Returns processed data
 
 =cut
 
 sub _filter_stack {
     my ($self, $data) = @_;
-    return $data if(numeric_level($data->{severity}) <= numeric_level('warn'));
+
+    return $data if (numeric_level($data->{severity}) <= numeric_level('warn'));
+
     # now severity > warn
     return $data if $self->{log_level} >= numeric_level('debug');
+
     delete $data->{stack};
     return $data;
 }
 
 =head2 _collapse_future_stack
 
-The future frames are too much and too tedious. This method will keep only one frame if there are many continuously future frames.
-Parameter: log data
-Return: log data
+The future L<FUTURE> frames are too much and too tedious. This method will keep only one
+frame if there are many continuously future frames.
+
+Takes the following arguments as named parameters
+
+=over 4
+
+=item * C<$data> Log stack data
+
+=back
+
+Returns log data
 
 =cut
 
-sub _collapse_future_stack{
+sub _collapse_future_stack {
     my ($self, $data) = @_;
     my $stack = $data->{stack};
     my @new_stack;
     my $previous_is_future;
-    for my  $frame ($stack->@*){
-        if($frame->{package} eq 'Future'){
-            next if($previous_is_future);
+
+    for my $frame ($stack->@*) {
+        if ($frame->{package} eq 'Future') {
+            next if ($previous_is_future);
             push @new_stack, $frame;
             $previous_is_future = 1;
-        }
-        else{
+        } else {
             push @new_stack, $frame;
             $previous_is_future = 0;
         }
     }
     $data->{stack} = \@new_stack;
+
     return $data;
 }
 
+=head2 _fh_is_tty
+
+Check the filehandle opened to tty
+Returns boolean
+
+=cut
+
 sub _fh_is_tty {
     my $fh = shift;
-   return -t $fh;
+
+    return -t $fh;
 }
 
+=head2 _in_container
+
+Checks in the container
+Returns boolean
+
+=cut
+
 sub _in_container {
     return -r '/.dockerenv';
 }
 
 =head2 _linux_flock_data
 
-Param: lock type. It can be F_WRLCK or F_UNLCK
+=over 4
+
+=item * C<$type> lock type - F_WRLCK or F_UNLCK
 
-return: A FLOCK structure
+=back
+
+Returns a FLOCK structure
```

Comment:
what's a `FLOCK structure`?

Reply:
Sorry it is a linux `struct flock`  structure used to describe the file lock type